keiko-redis/src/main/kotlin/com/netflix/spinnaker/q/redis/RedisQueue.kt

```diff
@@ -176,35 +177,34 @@ class RedisQueue(
 
           fingerprints.forEach { fingerprint ->
             val attempts = redis.hgetInt(attemptsKey, fingerprint)
-            if (attempts >= Queue.maxRetries) {
```

Comment:
This resulted in most long running tasks getting deadlettered whenever they ended up unacked, as `attemptsKey` was incremented on every message read. The `attempts` message attribute continues to be incremented as before, but `attemptsKey` is now only incremented by the unacked message retrier, or when an `IOException` is thrown in `readMessage()`.